fiftyone/server/routes/get_similar_labels_frames.py

```diff
@@ -0,0 +1,312 @@
+                                                },
+                                            }
+                                        },
+                                        "as": "d",
+                                        "in": "$$d._id",
+                                    }
+                                },
+                                "frameNumber": 1,
+                            }
+                        },
+                    ],
+                }
+            },
+            # Stage 3:
+            # Merge the results from the "rootLevel" and "nested" pipelines
+            # into one combined array.
+            {
+                "$project": {
+                    "combined": {"$concatArrays": ["$rootLevel", "$nested"]}
+                }
+            },
+            # Stage 4:
+            # Transform the combined array of label documents into a mapping
+            # object.
+            # - Use $reduce to iterate over each element and build an array
+            #   of key-value pairs.
+            # - Each key-value pair maps the label's _id (converted to a
+            #   string) to the corresponding frame number.
+            # - Finally, use $arrayToObject to convert the array of key-value
+            #   pairs into a single object.
+            {
+                "$project": {
+                    "labelIdMap": {
+                        "$arrayToObject": {
+                            "$reduce": {
+                                "input": "$combined",
+                                "initialValue": [],
+                                "in": {
+                                    "$concatArrays": [
+                                        "$$value",
+                                        {
+                                            "$map": {
+                                                "input": "$$this.labelIds",
+                                                "as": "labelId",
+                                                "in": {
+                                                    "k": {
+                                                        "$toString": "$$labelId"
+                                                    },
+                                                    "v": "$$this.frameNumber",
+                                                },
+                                            },
+                                        },
+                                    ],
+                                },
+                            },
+                        },
+                    },
+                }
+            },
+        ]
+
+        pipeline = view._pipeline(
+            frames_only=True, support=support, post_pipeline=post_pipeline
+        )
+
+        conn = foo.get_db_conn()
+        collection = conn[view._dataset._sample_collection_name]
+        label_id_map = list(foo.aggregate(collection, pipeline))[0][
+            "labelIdMap"
+        ]
```

Comment:
This needs to run with an async call either with `fiftyone.core.utils.run_sync_task` or `foo.get_async_db_conn()`. Currently this aggregate call will lock up the entire server waiting on MongoDB io

Reply:
good flag, fixed